frontend/src/v5/ui/routes/dashboard/projects/calibration/calibrationContext.tsx

```diff
@@ -29,8 +32,14 @@ export interface CalibrationContextType {
 	isCalibrating: boolean;
 	origin: string;
 	setOrigin: (origin: string) => void;
+	isCalibrating3D: boolean,
+	setIsCalibrating3D: (isCalibrating3D: boolean) => void;
+	vector3D: Vector3D,
+	setVector3D: React.Dispatch<React.SetStateAction<Vector3D>>;
```

Comment:
Does this need to be like this?

Reply:
I copied the typescript definition of `setState` so that I can call this function with both these syntaxes:
- `setState(newState)`
- `setState((oldState) => { /** code... */ ; return newState; })`

I try to simplify it though